plugins/inputs/burrow/burrow.go

```diff
@@ -0,0 +1,349 @@
+	u := fmt.Sprintf("%s/v2/kafka", baseUrl.String())
+	resp, err := b.client.Get(u)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	clusterList := &BurrowResponseClusterList{}
+	err = json.NewDecoder(resp.Body).Decode(clusterList)
+	if err != nil {
+		return nil, err
+	}
+
+	return clusterList.Clusters, nil
+}
+
+func (b *Burrow) getTopicList(baseUrl *url.URL, cluster string) ([]string, error) {
+	u := fmt.Sprintf("%s/v2/kafka/%s/topic", baseUrl.String(), cluster)
+	resp, err := b.client.Get(u)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	topicList := &BurrowResponseTopicList{}
+	err = json.NewDecoder(resp.Body).Decode(topicList)
+	if err != nil {
+		return nil, err
+	}
+
+	return topicList.Topics, nil
+}
+
+func (b *Burrow) gatherTopicOffsets(baseUrl *url.URL, cluster string, topic string, acc telegraf.Accumulator) error {
+	u := fmt.Sprintf("%s/v2/kafka/%s/topic/%s", baseUrl.String(), cluster, topic)
+	resp, err := b.client.Get(u)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	topicDetail := &BurrowResponseTopicDetail{}
+	err = json.NewDecoder(resp.Body).Decode(topicDetail)
+	if err != nil {
+		return err
+	}
+
+	for i, offset := range topicDetail.Offsets {
+		tags := map[string]string{
+			"cluster":   cluster,
+			"topic":     topic,
+			"partition": strconv.Itoa(i),
+		}
+		fields := map[string]interface{}{
+			"offset": offset,
+		}
+		acc.AddFields("burrow_topic", fields, tags)
+	}
+
+	return nil
+}
+
+func (b *Burrow) getConsumerList(baseUrl *url.URL, cluster string) ([]string, error) {
+	u := fmt.Sprintf("%s/v2/kafka/%s/consumer", baseUrl.String(), cluster)
+	resp, err := b.client.Get(u)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	consumerList := &BurrowResponseConsumerList{}
+	err = json.NewDecoder(resp.Body).Decode(consumerList)
+	if err != nil {
+		return nil, err
+	}
+
+	return consumerList.Consumers, nil
+}
+
+func (b *Burrow) gatherConsumerStatus(baseUrl *url.URL, cluster, group string, acc telegraf.Accumulator) error {
+	u := fmt.Sprintf("%s/v2/kafka/%s/consumer/%s/lag", baseUrl.String(), cluster, group)
+	resp, err := b.client.Get(u)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	consumerStatus := &BurrowResponseConsumerStatus{}
+	err = json.NewDecoder(resp.Body).Decode(consumerStatus)
+	if err != nil {
+		return err
+	}
+
+	for _, partition := range consumerStatus.Status.Partitions {
+		tags := map[string]string{
+			"cluster":   cluster,
+			"group":     group,
+			"topic":     partition.Topic,
+			"partition": strconv.Itoa(int(partition.Partition)),
+			"status":    partition.Status,
+		}
+
+		startFields := map[string]interface{}{
+			"offset": partition.Start.Offset,
+			"lag":    partition.Start.Lag,
+		}
+		acc.AddFields("burrow_consumer", startFields, tags, time.Unix(0, partition.Start.Timestamp*int64(time.Millisecond)))
+
+		endFields := map[string]interface{}{
```

Comment:
This seems problematic to me, since we are using the same tagset and fields as the start offset/lag, and there is potential for overwrite since we are storing the values at a timestamp based on the partition start/end.